maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/derive/schema/DeriveSchema.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.maven.derive.schema;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.confluent.kafka.schemaregistry.utils.JacksonMapper;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.Map;
+
+public abstract class DeriveSchema {
+
+  protected final HashMap<String, String> classToDataType = new HashMap<>();
+  protected static final ObjectMapper mapper = JacksonMapper.INSTANCE;
+
+  ArrayList<ObjectNode> getSchemaOfAllElements(List<JsonNode> messages, String name)
+      throws JsonProcessingException {
+    // Get schema of each message based on type
+    ArrayList<ObjectNode> schemaList = new ArrayList<>();
+    for (int i = 0; i < messages.size(); i++) {
+      try {
+        schemaList.add(getSchemaOfElement(messages.get(i), name));
+      } catch (IllegalArgumentException e) {
+        throw new IllegalArgumentException(
+            String.format("unable to find schema for message %d: %s", i, messages.get(i)), e);
+      }
+    }
+    return schemaList;
+  }
+
+  private ObjectNode getSchemaOfElement(JsonNode message, String name)
+      throws JsonProcessingException {
+    // Get schema of message based on type
+    Optional<ObjectNode> primitiveSchema = getPrimitiveSchema(message);
+    if (primitiveSchema.isPresent()) {
+      return primitiveSchema.get();
+    } else if (message instanceof ArrayNode) {
+      return getSchemaForArray(DeriveSchemaUtils.getListFromArray((ArrayNode) message), name);
+    } else {
+      ObjectNode objectNode = mapper.valueToTree(message);
+      return getSchemaForRecord(objectNode);
+    }
+  }
+
+  public Optional<ObjectNode> getPrimitiveSchema(JsonNode field) {
+    // Generate Schema for Primitive data types
+    String inferredType = field.getClass().getName();
+    if (classToDataType.containsKey(inferredType)) {
+      try {
+        String schema = String.format("{\"type\":\"%s\"}", classToDataType.get(inferredType));
```

Comment:
nit: this format should be defined in probably in Derive Schema utils  as a public static constant.